Crashes when using --git-parallel

While running betterleaks it crashes for many projects with a similar error

```sh
Command '['betterleaks', 'git', '.project', '--report-format', 'json', '--report-path', PosixPath('/builds/<censored>'), '--config', 'gitleaks.toml', '--git-workers', '8']' returned non-zero exit status 2.


  ○
  ○
  ●
  ○  betterleaks 1.1.2

[90m11:54PM[0m [32mINF[0m [1mUnknown SCM platform. Use --platform to include links in findings.[0m [36mhost=[0mcode.siemens.com
[90m11:54PM[0m [32mINF[0m [1mparallel git scan[0m [36mchunk_size=[0m2 [36mcommits=[0m10 [36mworkers=[0m8
panic: runtime error: slice bounds out of range [12:10]
goroutine 1 [running]:
github.com/betterleaks/betterleaks/sources.(*ParallelGit).Fragments(0xf775b699500, {0x1371d20, 0x20026e0}, 0xf775c7f2040)
	/Users/zrice/code/betterleaks/betterleaks/sources/parallel_git.go:73 +0x399
github.com/betterleaks/betterleaks/detect.(*Detector).DetectSource(0xf775b7062c8, {0x1371d20, 0x20026e0}, {0x1365660, 0xf775b699500})
	/Users/zrice/code/betterleaks/betterleaks/detect/detect.go:331 +0x18b
github.com/betterleaks/betterleaks/cmd.runGit(0x1fd0300, {0xf775b635dd0, 0x1, 0x12aaec9?})
	/Users/zrice/code/betterleaks/betterleaks/cmd/git.go:112 +0x6bd
github.com/spf13/cobra.(*Command).execute(0x1fd0300, {0xf775b635d40, 0x9, 0x9})
	/Users/zrice/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1019 +0xafb
github.com/spf13/cobra.(*Command).ExecuteC(0x1fcfa60)
	/Users/zrice/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1148 +0x465
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/zrice/go/pkg/mod/github.com/spf13/cobra@v1.10.1/command.go:1071
github.com/betterleaks/betterleaks/cmd.Execute()
	/Users/zrice/code/betterleaks/betterleaks/cmd/root.go:297 +0x1e
main.main()
	/Users/zrice/code/betterleaks/betterleaks/main.go:18 +0x9
```

--- a/sources/parallel_git.go
+++ b/sources/parallel_git.go
@@ -69,6 +69,9 @@ func (s *ParallelGit) Fragments(ctx context.Context, yield FragmentsFunc) error 
 	g, gctx := errgroup.WithContext(ctx)
 	for i := range workers {
 		start := i * chunkSize
+		if start >= count {
+			break
+		}
 		end := min(start+chunkSize, count)
 		chunk := commits[start:end]
 		g.Go(func() error {

--- a/sources/parallel_git_test.go
+++ b/sources/parallel_git_test.go
@@ -5,6 +5,7 @@ import (
 	"os"
 	"os/exec"
 	"path/filepath"
+	"strconv"
 	"strings"
 	"sync"
 	"testing"
@@ -184,6 +185,89 @@ func TestParallelGitSingleCommit(t *testing.T) {
 
 	if len(c.list) != 1 {
 		t.Errorf("got %d fragments, want 1", len(c.list))
+	}
+}
+
+// TestParallelGitWorkersExceedChunks reproduces the panic from issue where
+// workers * chunkSize > commit count (e.g. 10 commits, 8 workers, chunkSize=2
+// means worker 6 would slice [12:10]).
+func TestParallelGitWorkersExceedChunks(t *testing.T) {
+	dir := initTestRepo(t)
+
+	// Create exactly 10 commits to reproduce the original bug scenario
+	for i := range 10 {
+		commitFile(t, dir, "file.txt", strings.Repeat("x", i+1)+"\n", "commit "+strconv.Itoa(i))
+	}
+
+	ctx := context.Background()
+	src := &ParallelGit{
+		RepoPath: dir,
+		Config:   &config.Config{},
+		Sema:     semgroup.NewGroup(ctx, 10),
+		Workers:  8, // 10 commits / 8 workers = chunkSize 2, only 5 chunks needed
+	}
+
+	var c collectFragments
+	err := src.Fragments(ctx, c.yield)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(c.list) != 10 {
+		t.Errorf("got %d fragments, want 10", len(c.list))
+	}
+}
+
+// TestParallelGitWorkersEqualCommits verifies the edge case where workers == commits.
+func TestParallelGitWorkersEqualCommits(t *testing.T) {
+	dir := initTestRepo(t)
+
+	for i := range 3 {
+		commitFile(t, dir, "f"+strconv.Itoa(i)+".txt", "data\n", "c"+strconv.Itoa(i))
+	}
+
+	ctx := context.Background()
+	src := &ParallelGit{
+		RepoPath: dir,
+		Config:   &config.Config{},
+		Sema:     semgroup.NewGroup(ctx, 10),
+		Workers:  3,
+	}
+
+	var c collectFragments
+	err := src.Fragments(ctx, c.yield)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(c.list) != 3 {
+		t.Errorf("got %d fragments, want 3", len(c.list))
+	}
+}
+
+// TestParallelGitManyWorkersFewerCommits ensures no panic when workers far exceed commits.
+func TestParallelGitManyWorkersFewerCommits(t *testing.T) {
+	dir := initTestRepo(t)
+
+	commitFile(t, dir, "a.txt", "data\n", "c1")
+	commitFile(t, dir, "b.txt", "data\n", "c2")
+
+	ctx := context.Background()
+	src := &ParallelGit{
+		RepoPath: dir,
+		Config:   &config.Config{},
+		Sema:     semgroup.NewGroup(ctx, 10),
+		Workers:  100, // far more workers than commits
+	}
+
+	var c collectFragments
+	err := src.Fragments(ctx, c.yield)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if len(c.list) != 2 {
+		t.Errorf("got %d fragments, want 2", len(c.list))
 	}
 }
 

